ui/src/app/edge/live/common/grid/modal/modal.ts

```diff
@@ -1,53 +1,146 @@
-        this.meters.push({ component: component, isAsymmetric: isAsymmetric });
-      })
-
-    channelAddresses.push(
-      new ChannelAddress("_sum", 'GridMode'),
-      new ChannelAddress('_sum', 'GridActivePower'),
-      new ChannelAddress('_sum', 'GridActivePowerL1'),
-      new ChannelAddress('_sum', 'GridActivePowerL2'),
-      new ChannelAddress('_sum', 'GridActivePowerL3'),
-    )
-    return channelAddresses;
+    this.edge.getConfig(this.websocket).pipe(filter(config => !!config)).subscribe((config) => {
+      this.fields = ModalComponent.generateGridModal(this.edge.id, config, this.edge.role, this.translate);
+    })
+    return [];
   }
 
-  protected override onCurrentData(currentData: CurrentData): void {
+  public static generateModalMeterPhases(component: EdgeConfig.Component, translate: TranslateService, userRole: Role): FormlyFieldLine[] {
+    let fields: FormlyFieldLine[] = [];
 
-    this.grid.mode = currentData.allComponents["_sum/GridMode"]
-    let gridActivePower = currentData.allComponents['_sum/GridActivePower']
-    this.grid.buyFromGrid = gridActivePower > 0 ? gridActivePower : 0;
-    this.grid.sellToGrid = gridActivePower < 0 ? (gridActivePower * -1) : 0;
+    for (let phase of ['L1', 'L2', 'L3']) {
+      fields.push(
+        {
+          type: 'line',
+          name: translate.instant('General.phase') + " " + phase + " ",
+          nameSuffix: Utils.ADD_NAME_SUFFIX_FOR_GRIDSELL_OR_GRIDBUY(translate),
+          indentation: TextIndentation.SIMPLE,
+          channel: component.id + '/ActivePower' + phase,
+          children: [
+            Role.isAtLeast(userRole, Role.INSTALLER) && {
+              type: 'line-item',
+              channel: component.id + '/Voltage' + phase,
+              converter: Utils.CONVERT_TO_VOLT,
+              indentation: TextIndentation.SIMPLE,
+            },
+            Role.isAtLeast(userRole, Role.INSTALLER) && {
+              type: 'line-item',
+              channel: component.id + '/Current' + phase,
+              converter: Utils.CONVERT_TO_CURRENT,
+              indentation: TextIndentation.SIMPLE,
+            },
+            {
+              type: 'line-item',
+              channel: component.id + '/ActivePower' + phase,
+              converter: Utils.CONVERT_TO_GRIDSELL_OR_GRIDBUY_POWER,
+              indentation: TextIndentation.SIMPLE
+            }
+          ]
+        })
+    };
+
+    return fields;
   }
 
-  protected setTranslatedName = (power: number | null) => {
-    if (power == null || power == 0) {
-      return "";
+  public static generateGridModal(edgeId: string, edgeConfig: EdgeConfig, userRole: Role, translate: TranslateService): FormlyFieldConfig[] {
+
+    const hasMultipleGridMeters = edgeConfig.getComponentsImplementingNature("io.openems.edge.meter.api.AsymmetricMeter")
+      .filter(comp => comp.isEnabled && edgeConfig.isTypeGrid(comp)).length > 1;
```

Comment:
You should store the list of `AsymmetricMeter`s here, then you can use it again later